src/visual/Paper.js

```diff
@@ -43,12 +43,16 @@ export default class Paper extends joint.dia.Paper {
           return false;
         }
         // source
-        if (!magnetS || magnetS.getAttribute('class') !== 'port-body' ||
+        if (!magnetS || (magnetS.getAttribute('class') !== 'port-body' &&
+          magnetS.getAttribute('class') !== 'port-body-empty' &&
+          magnetS.getAttribute('class') !== 'port-body-empty available-magnet') ||
           magnetS.getAttribute('port-group') !== 'out') {
           return false;
         }
         // target
-        if (!magnetT || magnetT.getAttribute('class') !== 'port-body' ||
+        if (!magnetT || (magnetT.getAttribute('class') !== 'port-body' &&
+          magnetT.getAttribute('class') !== 'port-body-empty' &&
+          magnetT.getAttribute('class') !== 'port-body-empty available-magnet') ||
```

Comment:
Is it possible to write the condition in a more compact and error-proof way? I'm a bit disappointed by multiple `getAttribute()` calls and strict comparisons. What if there are two spaces between classes? different class order? one more (irrelevant) class that we decide to add later? The condition will be broken in this case and we'll waste our time on debugging.